web/src/components/people/TrainingForm.tsx

```diff
@@ -0,0 +1,136 @@
+  ssr: false,
+});
+const AlertMessage = dynamic(() => import("uikit-react-public").then((mod) => mod.Alert.Message), {
+  ssr: false,
+});
+
+const TrainingKindOptions = {
+  //  is there a better way of doing this? Won't let me use type as a value
+  nhsd: "training_kind_nhsd",
+};
+
+type TrainingFormProps = {
+  id: string;
+  setTrainingDialogOpen: (name: boolean) => void;
+  updatePersonUI: (id: string, training?: TrainingRecord) => void;
+};
+
+export default function TrainingForm(TrainingFormProps: TrainingFormProps) {
+  const { id, setTrainingDialogOpen, updatePersonUI } = TrainingFormProps;
+  const [errorMessage, setErrorMessage] = useState<string | null>(null);
+  const [errorType, setErrorType] = useState<AlertType>("warning");
+
+  const [trainingKind, setTrainingKind] = useState<TrainingKind | null>(null);
+  const [trainingDisplayDate, setTrainingDisplayDate] = useState<string | undefined>(undefined);
+  const [trainingDate, setTrainingDate] = useState<string>("");
+
+  const closeDialog = () => {
+    setTrainingDialogOpen(false);
+  };
+
+  const handleSubmit = async (event: FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (!trainingKind) return setErrorMessage("Please provide a training kind.");
+    if (!trainingDisplayDate) return setErrorMessage("Please enter the date the training was completed.");
+
+    try {
+      const response = await postPeopleById({
+        path: { id: id },
+        body: { training_kind: trainingKind, training_date: trainingDate },
+      });
+      if (!response.response.ok) throw new Error(`HTTP error! status: ${response.response.status}`);
+
+      closeDialog();
+
+      // matching with db value so training_kind_nhsd shows as nhsd
+      const trainingKindKey = Object.entries(TrainingKindOptions).find(([, value]) => value === trainingKind)?.[0];
+
+      updatePersonUI(id, {
+        kind: trainingKindKey as TrainingKind,
+        completed_at: trainingDisplayDate,
+        is_valid: true,
+      });
+    } catch (error) {
+      console.error("There was a problem submitting your request:", error);
+      setErrorMessage("Failed to submit agreement update. Please try again.");
+      setErrorType("error");
+    }
+  };
+
+  const setDateToToday = () => {
+    const today = new Date().toISOString();
+    setTrainingDisplayDate(today.split("T")[0]);
+    setTrainingDate(today);
+  };
+
+  return (
+    <AdminDialog setDialogOpen={setTrainingDialogOpen} data-cy="training">
+      <form onSubmit={handleSubmit} noValidate className={styles.form}>
+        <p>Use this form to validate a training certificate. Make sure you check the date on the certificate.</p>
+
+        <select
+          id="training_kind"
+          name="training_kind"
+          value={trainingKind || ""}
+          onChange={(e) => {
+            setTrainingKind(e.target.value as TrainingKind);
+            if (errorMessage) setErrorMessage(null);
```

Comment:
Am I reading this correctly? Is the error message being set to `null` if an error is found?

Reply:
think I was trying to make it so if there's an error message and then the user is typing, clear the error